TROUBLESHOOTING.md

```diff
@@ -0,0 +1,42 @@
+# Troubleshooting gRPC
+
+This guide is for troubleshooting gRPC implementations based on C core library (sources for most of them are living in the `grpc/grpc` repository).
+
+## Enabling extra logging and tracing
+
+Extra logging can be very useful for diagnosing problems. All gRPC implementations based on C core library support
+the `GRPC_VERBOSITY` and `GRPC_TRACE` environment variables that can be used to increase the amount of information
+that gets printed to stderr.
+
+## GRPC_VERBOSITY
+
+`GRPC_VERBOSITY` is used to set the minimum level of log messages printed by gRPC (supported values are `DEBUG`, `INFO` and `ERROR`).
+
+## GRPC_TRACE
+
+`GRPC_TRACE` can be used to enable extra logging for some internal gRPC components. Enabling the right traces can be invaluable
+for diagnosing for what is going wrong when things aren't working as intended. Possible values for GRPC_TRACE are listed in [Environment Variables Overview](doc/environment_variables.md).
+Multiple traces can be enable at once (use comma as separator).
+
+```
+# Enable debug logs for an application
+GRPC_VERBOSITY=debug ./helloworld_application_using_grpc
+```
+
+```
+# Print info about invocations of low-level C core API.
+# Note that trace logs of log level DEBUG won't be displayed
+# as GRPC_VERBOSITY and GRPC_TRACE are 2 independent settings.
+GRPC_TRACE=api ./helloworld_application_using_grpc
```

Comment:
Setting trace w/o verbosity does not accomplish much, since all the tracers use the INFO level of printing.

To test this, I ran
```
GRPC_TRACE=all bins/opt/h2_full_test simple_request
```
And nothing was output